New ConVars and behavior.

vertiKarl opened the pull request:
# Disclaimer
Needs more testing. We'll try it in our TTT party later today.

# Implements features requested in
  - #4 
  - #7 

# New ConVars
  - ttt2_dep_shared_death: Should the Deputy die with their Sheriff (Def: yes)
  - ttt2_dep_shared_killer: Should the killer be the same for the Sheriff and Deputy? (Def: yes)
  - ttt2_dep_custom_logic: allows using new custom logic (Def: no)
    - ttt2_dep_kill_logic: Specify what happens when this logic triggers (Def: target dies)
    - ttt2_dep_kill_logic_team: Specify which team this logic triggers for (Def: traitor)

# Changes
  - Moved Deagle refill settings to Deagle menu instead of role menu
  - Moved from PostPlayerDeath Hook to TTT2PostPlayerDeath for more information on the players involved.

# Fixes
  - Spelling:
    - "Deputys" → "Deputies"
    - "binded" → "bound"

vertiKarl committed:
New ConVars and behavior.

Implements features requested in:
  - #4
  - #7

- New ConVars
  - ttt2_dep_shared_death: Should the Deputy die with their Sheriff (Def: yes)
  - ttt2_dep_shared_killer: Should the killer be the same for the Sheriff and Deputy? (Def: yes)
  - ttt2_dep_custom_logic: allows using new custom logic (Def: no)
    - ttt2_dep_kill_logic: Specify what happens when this logic triggers (Def: target dies)
    - ttt2_dep_kill_logic_team: Specify which team this logic triggers for (Def: traitor)

- Changes
  - Moved Deagle refill settings to Deagle menu instead of role menu
  - Moved from PostPlayerDeath Hook to TTT2PostPlayerDeath for more information on the players involved.

- Fixes
  - Spelling:
    - "Deputys" → "Deputies"
    - "binded" → "bound"

vertiKarl committed:
removed unused variable declaration

vertiKarl committed:
made ConVars shared, so client doesn't throw errors when getting them

vertiKarl commented:
Worked for our last couple of sessions quite well.
We used it like this:
- ttt2_dep_shared_death: 1 (deputy dies when sheriff dies)
- ttt2_dep_shared_killer: 1 (if the deputy dies via the link with the sheriff, they die by the same player)
- ttt2_dep_custom_logic: 1 (enabled)
  - ttt2_dep_kill_logic: 1 (user dies)
  - ttt2_dep_kill_logic_team: 1 (any evil team)

TimGoll committed:
cleanup; small changes

TimGoll commented:
I gave it a try and changed a few minor things but there is one thing I'd like you to change:

![image](https://github.com/user-attachments/assets/2786cc58-46a4-45f0-8f1a-1570cd53897c)

- What is custom logic? Is this convar needed at all? If it is, it should have a more fitting label or a helpbox explaining what happens
- The text in front of the combo box is too long
- combobox entries should not have a capitalized word in second place and no dot in the end

vertiKarl committed:
- Added help box to deputy deagle config menu
- FIxed formatting for combobox

vertiKarl commented:
Thanks for taking a look!

> What is custom logic? Is this convar needed at all? If it is, it should have a more fitting label or a helpbox explaining what happens
- Custom logic is used by the server admin to override the existing logic when a player shoots another player with the deputy deagle. Ideally this could be quite expandable, allowing players to set up all kinds of behaviors, but at the moment it is quite basic. I have added a helpbox explaining a bit what it does. (Note: Setting it to "Nothing" does not disable custom logic but instead it stops the deputy deagle from working for that target team. Example: Sheriff shoots traitor, deagle gets used but the traitor does not change their team). To make this clearer I changed the nothing option to include "(not converting)".
> The text in front of the combo box is too long
- changed
> combobox entries should not have a capitalized word in second place and no dot in the end
- changed

EDIT: This is what it looks like now:
![image](https://github.com/user-attachments/assets/66533044-a63c-4cfe-92df-d8d88ce95b99)

TimGoll commented on lua/terrortown/lang/en/sheriff.lua:
```diff
@@ -20,3 +20,22 @@ L["label_dep_protection_time"] = "Protection time for new Deputy"
+(Default: Everyone shot with the deputy deagle converts to a deputy no matter what.)
+This allows you to change what happens when you shoot another player depending on which team they are on.
+A possible example: you shoot a traitor with the deputy deagle → you die.
+]]
```
```suggestion
L["label_ttt2_dep_custom_logic_help"] = [[Enable this if you want to override the default behavior.
(Default: Everyone shot with the deputy deagle converts to a deputy no matter what.)

This allows you to change what happens when you shoot another player depending on which team they are on.
A possible example: you shoot a traitor with the deputy deagle → you die.]]
```

TimGoll reviewed (commented):
If you change this for the German language as well and make sure that you use spaces instead of tabs (TTT2 uses spaces, you can also use our stylua profile if you want, that takes care of this automatically), then it is perfect imho.

vertiKarl committed:
Update lua/terrortown/lang/en/sheriff.lua

Co-authored-by: Tim Goll <github@timgoll.de>

vertiKarl committed:
Removed unnecessary space at end help box

vertiKarl committed:
Changed German translation.
Fixed typos.

vertiKarl commented:
Okay, hope all is well now!

I removed the trailing newline at the end of the help box and fixed a typo in the German translation.
And I also tried running the lua files through stylua formatter. As I am using the formatter for the first time I'm not quite sure if it even did anything? Seems like everything was already using spaces after your commit.

If there is anything else, let me know.

This is what it looks like in German:
![image](https://github.com/user-attachments/assets/fc3c989b-5d78-418d-b764-c8fb4ff2689f)

TimGoll reviewed (approved):
LGTM thank you. I'll try to push this to the workshop later today

vertiKarl commented:
Just wanted to mention, that in case you forgot, this has not been pushed to the workshop yet.
